src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -64,27 +67,41 @@ internal class WorkloadInstallCommand : CommandBase
             _downloadToCacheOption = parseResult.GetValueForOption(WorkloadInstallCommandParser.DownloadToCacheOption);
             _workloadIds = workloadIds ?? parseResult.GetValueForArgument(WorkloadInstallCommandParser.WorkloadIdArgument).ToList().AsReadOnly();
             _verbosity = parseResult.GetValueForOption(WorkloadInstallCommandParser.VerbosityOption);
-            _dotnetPath = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath);
+            _dotnetPath = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath)!;
             _userProfileDir = userProfileDir ?? CliFolderPathCalculator.DotnetUserProfileFolderPath;
             _sdkVersion = WorkloadOptionsExtensions.GetValidatedSdkVersion(parseResult.GetValueForOption(WorkloadInstallCommandParser.VersionOption), version, _dotnetPath, _userProfileDir);
-            _sdkFeatureBand = new SdkFeatureBand(_sdkVersion);
-            _tempDirPath = tempDirPath ?? (string.IsNullOrWhiteSpace(parseResult.GetValueForOption(WorkloadInstallCommandParser.TempDirOption)) ?
-                Path.GetTempPath() :
-                parseResult.GetValueForOption(WorkloadInstallCommandParser.TempDirOption));
+            _sdkFeatureBand = new(_sdkVersion);
             _fromRollbackDefinition = parseResult.GetValueForOption(WorkloadInstallCommandParser.FromRollbackFileOption);
+            
+            if (tempDirPath is null)
+            {
+                var fromOption = parseResult.GetValueForOption(WorkloadInstallCommandParser.TempDirOption);
+                _tempDirPath = string.IsNullOrWhiteSpace(fromOption) ? Path.GetTempPath() : fromOption;
+            }
+            else
+            {
+                _tempDirPath = tempDirPath;
+            }
+            
 
             var configOption = parseResult.GetValueForOption(WorkloadInstallCommandParser.ConfigOption);
             var sourceOption = parseResult.GetValueForOption(WorkloadInstallCommandParser.SourceOption);
-            _packageSourceLocation = string.IsNullOrEmpty(configOption) && (sourceOption == null || !sourceOption.Any()) ? null :
-                new PackageSourceLocation(string.IsNullOrEmpty(configOption) ? null : new FilePath(configOption), sourceFeedOverrides: sourceOption);
+            if (string.IsNullOrEmpty(configOption) && sourceOption != null && sourceOption.Any())
+            {
+                _packageSourceLocation = null;
+            }
+            else
+            {
+                _packageSourceLocation = new(string.IsNullOrEmpty(configOption) ? null : new FilePath(configOption), sourceFeedOverrides: sourceOption);
+            }
```

Comment:
That is not equivalent to the old code.  If `string.IsNullOrEmpty(configOption) && sourceOption == null`, then the old code would assign `_packageSourceLocation = null`, but your new code goes to the other branch instead.

Reply:
If you want to use `&&`, then `string.IsNullOrEmpty(configOption) && !(sourceOption != null && sourceOption.Any())` is equivalent to the old code, but I wouldn't say it's clearer than the old code. I feel there's a nice symmetry between string.IsNull***Or***Empty and `||`.

Reply:
yep, you are correct, I'm missing a ! around the &&, good catch.

Reply:
`||` does short-circuit.  `sourceOption == null || !sourceOption.Any()` does not call `sourceOption.Any()` if `sourceOption == null` is true.